rpcs3/Emu/Cell/Modules/cellAvconfExt.cpp

```diff
@@ -38,17 +110,17 @@ s32 cellVideoOutSetupDisplay()
 s32 cellAudioInGetDeviceInfo(u32 deviceNumber, u32 deviceIndex, vm::ptr<CellAudioInDeviceInfo> info)
 {
 	cellAvconfExt.todo("cellAudioInGetDeviceInfo(deviceNumber=0x%x, deviceIndex=0x%x, info=*0x%x)", deviceNumber, deviceIndex, info);
-	info->portType = CELL_AUDIO_IN_PORT_USB;
-	info->availableModeCount = 1;
-	info->state = CELL_AUDIO_IN_DEVICE_STATE_AVAILABLE;
-	info->deviceNumber = 0;
-	// Some games check if deviceId and type are the same.
-	info->deviceId = 0xDEADBEEF;
-	info->type = 0xBEEFDEAD;
-	info->availableModes[0].type = CELL_AUDIO_IN_CODING_TYPE_LPCM;
-	info->availableModes[0].channel = CELL_AUDIO_IN_CHNUM_NONE;
-	info->availableModes[0].fs = CELL_AUDIO_IN_FS_8KHZ | CELL_AUDIO_IN_FS_12KHZ | CELL_AUDIO_IN_FS_16KHZ | CELL_AUDIO_IN_FS_24KHZ | CELL_AUDIO_IN_FS_32KHZ | CELL_AUDIO_IN_FS_48KHZ;
-	strcpy(info->name, "USB Camera");
+
+	std::vector<device_in> ain_devices;
+	get_vec_input_devices(ain_devices);
```

Comment:
Why not just `std::vector<> ain_devices = get_input_devices()`? The compiler always avoids the copy as a basic optimization (Its part of the c++ compiler spec, copy ellision)